src/escrows/LPCurveYearnV2Escrow.sol

```diff
@@ -193,7 +230,8 @@ contract DolaFraxBPEscrow {
      * @dev Cannot deposit if there are Convex tokens in the escrow (only 1 strategy at a time)
      */
     function depositToYearn() external onlyBeneficiary {
-        if (stakedBalance > 0) withdrawFromConvex();
+        uint256 convexBalance = rewardPool.balanceOf(address(this));
+        if (convexBalance > 0) withdrawFromConvex();
         yearn.deposit(token.balanceOf(address(this)), address(this));
```

Comment:
Optional:
best practice for vault integrations is to check the `vault.availableDepositLimit()` such that you only deposit what the vault can allow, and not run into a revert.

In your case, nothing bad can happen. Worst case is a revert, without blocking critical code paths. Yearn's Curve LP vaults typically all have extremely high deposit limits that never get reached.

reference:
https://github.com/yearn/yearn-vaults/blob/97ca1b2e4fcf20f4be0ff456dabd020bfeb6697b/contracts/Vault.vy#L1605